public/app/features/explore/ExploreToolbarExtensionPoint/ExploreToolbarExtensionPoint.tsx

```diff
@@ -0,0 +1,87 @@
+import React, { lazy, ReactElement, Suspense, useMemo, useState } from 'react';
+
+import {
+  isPluginExtensionCommand,
+  isPluginExtensionLink,
+  PluginExtension,
+  PluginExtensionLink,
+  PluginExtensionPlacements,
+  SelectableValue,
+} from '@grafana/data';
+import { getPluginExtensions } from '@grafana/runtime';
+import { ButtonGroup, ButtonSelect } from '@grafana/ui';
+import { ExploreId } from 'app/types';
+
+import { ConfirmNavigationModal } from './ConfirmNavigationModal';
+
+const AddToDashboard = lazy(() =>
+  import('../AddToDashboard').then(({ AddToDashboard }) => ({ default: AddToDashboard }))
+);
+
+type Props = {
+  exploreId: ExploreId;
+};
+
+type ExploreExtensionPointContext = {};
+
+export function ExploreToolbarExtensionPoint(props: Props): ReactElement {
+  const { exploreId } = props;
+  const [extension, setExtension] = useState<PluginExtensionLink | undefined>();
+  const context = useExtensionPointContext(props);
+  const options = useExtensionsAsOptions(context);
+
+  if (options.length === 0) {
+    return (
+      <Suspense fallback={null}>
+        <AddToDashboard exploreId={exploreId} />
+      </Suspense>
+    );
+  }
+
+  return (
+    <Suspense fallback={null}>
+      <ButtonGroup>
+        <AddToDashboard key="add-to-dashboard" exploreId={exploreId} />
```

Comment:
Are you also planning to add more info to the context provided to the handler? For example list of queries in Explore? I wonder if it's feasible (and if it makes sense) to make `AddToDashboard` an extension like others 🤔

Reply:
One more related thing - if there's no plan for more context being provided to the extension, then not sure if that's the best placement for "global" Explore extensions. Toolbars are shown per split pane so it creates impressions that the action is somehow related to the single pane.

Reply:
> Are you also planning to add more info to the context provided to the handler? For example list of queries in Explore? I wonder if it's feasible (and if it makes sense) to make `AddToDashboard` an extension like others 🤔

Yes - I was actually thinking that is might be a natural follow up. To move the `addToDashboard` and register it as an extension.

Reply:
> One more related thing - if there's no plan for more context being provided to the extension, then not sure if that's the best placement for "global" Explore extensions. Toolbars are shown per split pane so it creates an impression that the action is somehow related to the single pane.

The plan is to add more information to the context and the extensions should be tied to the query. With I guess is per pane?